models/article-model.js

```diff
@@ -1,61 +1,66 @@
 const db = require("../db/connection");
-const {checkArticleExists} = require("../db/utils");
-
+const { checkArticleExists } = require("../db/utils");
 
 exports.fetchArticleByID = (articleID) => {
   return db
-    .query(`SELECT *FROM articles WHERE article_id =$1`,[articleID])
+    .query(`SELECT *FROM articles WHERE article_id =$1`, [articleID])
     .then(({ rows }) => {
-        const article = rows[0]
+      const article = rows[0];
       if (!article) {
         return Promise.reject({ status: 404, msg: "Not found" });
       }
-      return article
+      return article;
     })
     .catch((err) => {
-        throw err;
-      });
+      throw err;
+    });
 };
 
+exports.fetchAllArticles = (sort_by = "created_at", order = "DESC") => {
+  const validSortBys = ["title", "topic", "author", "votes", "created_at"];
+  const validOrderBys = ["ASC", "DESC"];
 
-exports.fetchAllArticles = () =>{
-    return db
-    .query(`SELECT articles.article_id, articles.title, articles.topic, articles.author, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.article_id)::INT AS comment_count FROM articles LEFT JOIN comments ON comments.article_id =articles.article_id
-        GROUP BY 
+  const query = `SELECT articles.article_id, articles.title, articles.topic, articles.author, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.article_id)::INT AS comment_count FROM articles LEFT JOIN comments ON comments.article_id =articles.article_id GROUP BY 
         articles.article_id, 
         articles.title, 
         articles.topic, 
         articles.author, 
         articles.created_at, 
         articles.votes, 
         articles.article_img_url
-      ORDER BY articles.created_at DESC;
-    `)
+        ORDER BY articles.${sort_by} ${order};`;
+
+  return db
+    .query(query)
     .then(({ rows }) => {
+      if (!validSortBys.includes(sort_by) || !validOrderBys.includes(order)) {
```

Comment:
You want to do this check before you run the SQL on your database, if there was any malicious code it would have ran by this point